nixos/modules/services/audio/hqplayerd.nix

```diff
@@ -0,0 +1,62 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.hqplayerd;
+  package = pkgs.hqplayerd;
+in {
+  options = {
+    services.hqplayerd = {
+      enable = mkEnableOption "HQPlayer Embedded";
+
+      licenseFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = ''
+          Path to the HQPlayer license key file.
+
+          Without this, the service will run in trial mode and restart every 30
+          minutes.
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Open TCP port 8088 in the firewall for the server.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.etc = {
+      "hqplayer/hqplayerd4-key.xml" = mkIf (cfg.licenseFile != null) { source = cfg.licenseFile; };
+      "modules-load.d/taudio2.conf".source = "${package}/etc/modules-load.d/taudio2.conf";
+    };
+
+    networking.firewall = mkIf cfg.openFirewall {
+      allowedTCPPorts = [ 8088 ];
+    };
+
+    services.udev.packages = [ package ];
+
+    systemd = {
+      packages = [ package ];
+
+      services.hqplayerd = {
+        wantedBy = [ "multi-user.target" ];
+        preStart = pkgs.writeShellScript "hqplayerd-pre-start" ''
+          # 2021-06-13
+          # yes, this sucks. No, there's no way to get around it.
+          # I've emailed the person who develops this asking for it to be
+          # possible to set make it use `/var/lib` instead of `/var`.
+          mkdir -p /var/hqplayer/{home,web}
```

Comment:
Oh so this software has to run as `root` too? 😞

Reply:
Yeah, it's pretty unfortunate. I'm trying to contact the developer to better understand what permissions it needs.

I know it needs `audio` group perms, but it also likes to set it's own prio to realtime, but I don't think it knows how to use `rtkit` and thus the `root` requirement :/

I'm honestly more OK with it running as `root`, since I can still just shove it into a NixOS container, than with it polluting `/var/`.

Another day another cursed thing to package!

Reply:
Ha. Ok cool 👍